scripts/apps/highlights/services/HighlightsService.ts

```diff
@@ -111,7 +114,25 @@ export function HighlightsService(api, $q, $cacheFactory, packages: IPackagesSer
      * Mark an item for a highlight
      */
     service.markItem = function(highlight, markedItem) {
-        return api.save('marked_for_highlights', {highlights: [highlight], marked_item: markedItem._id});
+        dispatchEvent(new CustomEvent('highlights-loading', {
```

Comment:
Can you abstract it into a function so if you want to use it for some other action you would only need to call a function, but wouldn't need to know about events?

```
function trackArticleActionProgress<T>(p: () => Promise<T>, successMessage: string, errorMessage: string): Promise<T> {
    //
}

function markItem(highlight, markedItem) {
    return trackArticleActionProgress(
        () => api.save(
            'marked_for_highlights',
            {highlights: [highlight], marked_item: markedItem._id},
        ),
        gettext('Item marked'),
        gettext('Couldn\'t mark item'),
    );
}

```